src/test_application/OSPMagnum.cpp

```diff
@@ -159,71 +161,195 @@ osp::active::ActiveScene& OSPMagnum::scene_create(std::string && name)
-
-    // Camera and Game controls, handled in DebugCameraController
-
-    // Switch to next vehicle
-    rUserInput.config_register_control("game_switch", false,
-            {{0, (int) Key_t::V, VarTrig_t::PRESSED, false, VarOp_t::OR}});
-
-    // Set UI Up/down/left/right to arrow keys. this is used to rotate the view
-    // for now
-    rUserInput.config_register_control("ui_up", true,
-            {{osp::sc_keyboard, (int) Key_t::Up, VarTrig_t::PRESSED, false, VarOp_t::AND}});
-    rUserInput.config_register_control("ui_dn", true,
-            {{osp::sc_keyboard, (int) Key_t::Down, VarTrig_t::PRESSED, false, VarOp_t::AND}});
-    rUserInput.config_register_control("ui_lf", true,
-            {{osp::sc_keyboard, (int) Key_t::Left, VarTrig_t::PRESSED, false, VarOp_t::AND}});
-    rUserInput.config_register_control("ui_rt", true,
-            {{osp::sc_keyboard, (int) Key_t::Right, VarTrig_t::PRESSED, false, VarOp_t::AND}});
-
-    rUserInput.config_register_control("ui_rmb", true,
-            {{osp::sc_mouse, (int) Mouse_t::Right, VarTrig_t::PRESSED, false, VarOp_t::AND}});
-
-    rUserInput.config_register_control("debug_planet_update", false,
-            {{0, (int) Key_t::LeftCtrl, VarTrig_t::HOLD, false, VarOp_t::AND},
-             {0, (int) Key_t::One, VarTrig_t::PRESSED, false, VarOp_t::OR}});
+	// Configure Controls
+
+	// It should be pretty easy to write a config file parser that calls these
+	// functions.
+
+	// It should be pretty easy to write a config file parser that calls these
+	// functions.
+
+	//Load toml
+	auto data = toml::parse("settings.toml");
+	std::cout << "found file " << std::endl;
+
+	using namespace osp;
+
+	using Key_t = OSPMagnum::KeyEvent::Key;
+	using Mouse_t = OSPMagnum::MouseEvent::Button;
+	using VarOp_t = ButtonVarConfig::VarOperator;
+	using VarTrig_t = ButtonVarConfig::VarTrigger;
+
+	UserInputHandler& rUserInput = rOspApp.get_input_handler();
+	try {
+		for (const auto& [k, v] : data.as_table()) {
+			std::cout << k << std::endl;
+			std::string primary = toml::find(v, "primary").as_string();
+			std::vector<ButtonVarConfig> vec = parse_control(primary);
+
+			std::string secondary = toml::find(v, "secondary").as_string();
+			std::vector<ButtonVarConfig> secondaryKeys = parse_control(secondary);
+
+			vec.insert(vec.end(), secondaryKeys.begin(), secondaryKeys.end());
+
+			bool invert = toml::find(v, "invert").as_boolean();
+
+			//Add user input
+			rUserInput.config_register_control(k, invert, vec);
+		}
+	}
+	catch (std::out_of_range oor) {
+		std::cout << oor.what() << std::endl;
+	}
 }
+
+
+//Map for all the keys
+//The tuple is in this order: device, number, and hold/pressed
+using namespace osp;
+
+using Key_t = OSPMagnum::KeyEvent::Key;
+using Mouse_t = OSPMagnum::MouseEvent::Button;
+using VarOp_t = ButtonVarConfig::VarOperator;
+using VarTrig_t = ButtonVarConfig::VarTrigger;
+
+typedef std::tuple<int, int> button_tuple;
```

Comment:
ButtonTuple_t, prefer using instead of typedef, and also use UserInputHandler::DeviceId (which should also be DeviceId_t) instead of int for the first one.